src/__tests__/site-apps.ts

```diff
@@ -330,7 +296,42 @@ describe('SiteApps', () => {
             siteAppsInstance.afterDecideResponse(response)
 
             expect(siteAppsInstance.loaded).toBe(true)
-            expect(siteAppsInstance.enabled).toBe(false)
+        })
+
+        it('does not load site destinations if consent is not given', () => {
+            posthog.config.opt_in_site_apps = true
+            posthog.consent.isOptedOut = () => true
+            const response = {
+                siteApps: [
+                    { id: '5', type: 'site_app', url: '/site_app/5' },
+                    { id: '6', type: 'site_destination', url: '/site_app/6' },
+                ],
+            } as DecideResponse
+
+            siteAppsInstance.afterDecideResponse(response)
+
+            expect(typeof assignableWindow['__$$ph_site_app_5_callback']).toBe('function')
+            expect(typeof assignableWindow['__$$ph_site_app_6_callback']).toBe('undefined')
+        })
+
+        it('load site destinations if consent is given at a later time', () => {
+            posthog.config.opt_in_site_apps = true
+            posthog.consent.isOptedOut = () => true
+            const response = {
+                siteApps: [
+                    { id: '5', type: 'site_app', url: '/site_app/5' },
+                    { id: '6', type: 'site_destination', url: '/site_app/6' },
+                ],
+            } as DecideResponse
+
+            siteAppsInstance.afterDecideResponse(response)
+
+            posthog.consent.isOptedOut = () => false
+
+            siteAppsInstance.loadIfEnabled()
+
+            expect(typeof assignableWindow['__$$ph_site_app_5_callback']).toBe('function')
+            expect(typeof assignableWindow['__$$ph_site_app_6_callback']).toBe('function')
```

Comment:
Is this `loadIfEnabled` correct? You're explicitly loading the site apps/destinations after consent. And they do load. However, you're calling the "load" function directly in the test. I don't expect users to call this function. 

We should test that opting in the way users opt in starts the site apps.

Reply:
updated the test to call the `opt_out_capturing` and `opt_in_capturing` functions instead.